workflows/iuf/operations/management-nodes-rollout/management-two-master-nodes-rollout.yaml

```diff
@@ -216,63 +175,95 @@ spec:
+                      echo -e "DEBUG File $temp_file is empty after command <cray artifacts get $bucket $path $temp_file>. Output from command is -\n\n$result"
+                    fi
                     metal_image=$(jq '.artifacts | map({"path": .link.path, "type": .type}) | .[] | select( .type == "application/vnd.cray.image.rootfs.squashfs") | .path ' < $temp_file)
-                    echo "INFO  Setting metal.server image to: $metal_image"
+                    if [ -z "$metal_image" ]; then
+                      echo "ERROR Metal image (rootfs.squashfs image) is not present in the output of <cray artifacts get $bucket $path>"
+                    else
+                      echo "DEBUG Setting metal.server image to: $metal_image"
+                    fi
                     kernel_image=$(jq '.artifacts | map({"path": .link.path, "type": .type}) | .[] | select( .type == "application/vnd.cray.image.kernel") | .path ' < $temp_file)
                     kernel_image=$(echo "$kernel_image" | tr -d '"')
-                    echo "INFO Setting kernel image to: $kernel_image"
+                    if [ -z "$kernel_image" ]; then
+                      echo "ERROR Kernel image is not present in the output of <cray artifacts get $bucket $path>"
+                    else
+                      echo "DEBUG Setting kernel image to: $kernel_image"
+                    fi
                     initrd_image=$(jq '.artifacts | map({"path": .link.path, "type": .type}) | .[] | select( .type == "application/vnd.cray.image.initrd") | .path ' < $temp_file)
                     initrd_image=$(echo "$initrd_image" | tr -d '"')
-                    echo "INFO  Setting initrd image to: $initrd_image"
+                    if [ -z "$initrd_image" ]; then
+                      echo "ERROR initrd image is not present in the output of <cray artifacts get $bucket $path>"
+                    else
+                      echo "DEBUG Setting initrd image to: $initrd_image"
+                    fi
                     METAL_SERVER=$(cray bss bootparameters list --hosts "${TARGET_XNAME}" --format json | jq '.[] |."params"' \
                     | awk -F 'metal.server=' '{print $2}' \
                     | awk -F ' ' '{print $1}')
                     NEW_METAL_SERVER=$metal_image
                     PARAMS=$(cray bss bootparameters list --hosts "${TARGET_XNAME}" --format json | jq '.[] |."params"' | \
                         sed "/metal.server/ s|${METAL_SERVER}|${NEW_METAL_SERVER}|" | \
                         tr -d \")
-                    cray bss bootparameters update --hosts "${TARGET_XNAME}" \
+                    result=$(cray bss bootparameters update --hosts "${TARGET_XNAME}" \
                       --kernel $kernel_image \
                       --initrd $initrd_image \
-                      --params "${PARAMS}"
-
+                      --params "${PARAMS}" 2>&1)
+                    if[ $? != 0 ]; then
```

Comment:
it would be nice to see an INFO message here when the result code is 0 that it successfully set the parameters for the host, including all the items it set.  like

"Successfully set cray bss bootparameters for host <host> with --kernel <kernel>, --initrd <initrd> and --params <params>" or something like that.